build-tools/babel/babel-loader-cache-identifier/index.js

```diff
@@ -28,6 +28,6 @@ module.exports = JSON.stringify( {
 	'babel-plugin-transform-wpcalypso-async': getModuleVersion(
 		'@automattic/babel-plugin-transform-wpcalypso-async'
 	),
-	babelrc: fs.readFileSync( path.resolve( __dirname, '../../../../../babel.config.js' ), 'utf8' ),
+	babelrc: fs.readFileSync( path.join( __dirname, '../../../babel.config.js' ), 'utf8' ),
```

Comment:
It's unfortunate that the _babel.config.js_ path is hardcoded like this. It should be an option passed by the consumer, which [resolves the file path just two lines earlier](https://github.com/Automattic/wp-calypso/blob/master/client/webpack.config.js#L211). We should fix that someday.

By the way, why the change from `resolve` to `join`? `join` produces paths like `build-tools/babel/../../babel.config.js`, while `resolve` produces `babel.config.js`. The original `resolve` looks more desirable.

Reply:
Good catch, this is a leftover from a test. I'll switch it back to be consistent with other usages.

For the record, both `path.resolve()` and `path.join()` produce the same result here. Both methods will normalize the result, which takes care of resolving `../` and using the correct path separator for the platform.

This code
```js
console.log( __dirname );
console.log( path.resolve( __dirname, '../../../babel.config.js' ) );
console.log( path.join( __dirname, '../../../babel.config.js' ) );
console.log( path.normalize( `${ __dirname }/../../../babel.config.js` ) );
console.log( path.win32.resolve( __dirname, '../../../babel.config.js' ) );
console.log( path.win32.join( __dirname, '../../../babel.config.js' ) );
console.log( path.win32.normalize( `${ __dirname }/../../../babel.config.js` ) );
```

Outputs this:
```
/Users/sergio/src/automattic/wp-calypso/build-tools/babel/babel-loader-cache-identifier
/Users/sergio/src/automattic/wp-calypso/babel.config.js
/Users/sergio/src/automattic/wp-calypso/babel.config.js
/Users/sergio/src/automattic/wp-calypso/babel.config.js
\Users\sergio\src\automattic\wp-calypso\babel.config.js
\Users\sergio\src\automattic\wp-calypso\babel.config.js
\Users\sergio\src\automattic\wp-calypso\babel.config.js
```

Reply:
Nice, I thought that `join` merely joins the path fragments together, making sure that path delimiters are inserted as needed.